.github/workflows/dart.yml

```diff
@@ -0,0 +1,55 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Dart
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      # Note: This workflow uses the latest stable version of the Dart SDK.
+      # You can specify other versions if desired, see documentation here:
+      # https://github.com/dart-lang/setup-dart/blob/main/README.md
+      # - uses: dart-lang/setup-dart@v1
+      - uses: dart-lang/setup-dart@9a04e6d73cca37bd455e0608d7e5092f881fd603
+
+      - name: Dart Preview fix
+        run: dart fix --dry-run
+        working-directory: app
+        
+      - name: Apply Fix
+        run: dart fix --apply
+        working-directory: app
+        
+      # Uncomment this step to verify the use of 'dart format' on each commit.
+      - name: Verify formatting
+        run: dart format --output=show . 
+        working-directory: app/lib
```

Comment:
we aren't really enforcing the formatting here, do we? We only format and show, but don't stop CI runs if they differ... But the latter is what we really want, no?

Reply:
Yes, CI pass will fail on formatting because of unformatted files in main branch. So we have to overwrite files and commit on it.